### PR TITLE
User Details

### PR DESCRIPTION
**Pull request by whyDontI**
Partially solves #22 

Added 
- GET /users
- GET /users/:id
- POST /users
- PATCH /users/:id

Changed
- divided routes/index.js into separate route files

### CODE REVIEW

File: controllers/membersController.js [hunk 1]
```diff
@@ -9,36 +10,87 @@ const db = require('../utils/firestore')
  */
 
 async function getMembers (req, res) {
-  const snapshot = await db.collection('members').get()
+  try {
```
Comment (ankurnarkhede) [inline controllers/membersController.js:13]: Can you move the DB interaction logic to a /models?

File: controllers/membersController.js [hunk 2]
```diff
@@ -9,36 +10,87 @@ const db = require('../utils/firestore')
...
+ * Add new member
+ *
+ * @param req {Object} - Express request object
+ * @param req.body {Object} - Member object
+ * @param res {Object} - Express response object
+ * @param next {Function} - Express middleware function
+ */
+async function addNewMember (req, res) {
+  try {
+    const memberRef = await db.collection('members').doc(req.body.id)
```
Comment (ankurnarkhede) [inline controllers/membersController.js:75]: How will we be creating a relationship between `users` collection and `members`? The data that this API is storing, part of it like github_id, github_display_name is already added in users collection.

File: middlewares/validators/member.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      twitter_id: joi.string().optional(),
+      instagram_id: joi.string().optional(),
+      site: joi.string().optional()
+    })
+
+    try {
+      await schema.validateAsync(req.body)
+      next()
+    } catch (error) {
+      res.boom.badRequest(error.details[0].message)
```
Comment (ankurnarkhede) [inline middlewares/validators/member.js:24]: - add `return`
- add log

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: middlewares/validators/member.js (hunk 3 above)
Comment (whyDontI) [reply middlewares/validators/member.js:24]: what should be returned here?

File: controllers/membersController.js (hunk 2 above)
Comment (whyDontI) [reply controllers/membersController.js]: the is_member boolean will take care of this part

**Review by ankushdharkar — APPROVED** [review]
Looks good to me

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: controllers/usersController.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+const getUsers = async (req, res) => {
+  try {
+    const allUsers = await userQuery.fetchUsers(req.query)
+
+    if (allUsers.length) {
+      res.json({
+        message: 'Users returned successfully!',
+        users: allUsers
+      })
+      return
```
[the remark this answers is not in the document]
Comment (whyDontI) [reply controllers/usersController.js:20]: We are not returning the `res.json`.
`res.json` send the response back
return is used to stop the execution.

Making the changes

File: models/users.js [hunk 5]
```diff
@@ -8,51 +8,103 @@ const firestore = require('../utils/firestore')
...
- * Adds or updates the user data
+ * Adds the user data
  *
  * @param userData { Object }: User data object to be stored in DB
  * @return {Promise<{isNewUser: boolean, userId: string}|{isNewUser: boolean, userId: string}>}
  */
-const addOrUpdate = async (userData) => {
-  let userInfo
-
+const addUser = async (userData) => {
```
[the remark this answers is not in the document]
Comment (whyDontI) [reply models/users.js:16]: Merging both functions as `addOrUpdate`

**Comment by ankurnarkhede** [comment]
LGTM.

**Review by ankurnarkhede — APPROVED** [review]

**Pull request merged**